File: BookWorm-NET/BookWormNET/Services/Implementation/EmailService.cs
using BookWormNET.dto;
using BookWormNET.Models;
using MailKit.Net.Smtp;
using MailKit.Security;
using Microsoft.Extensions.Options;
using MimeKit;

namespace BookWormNET.Services
{
    public class EmailService
    {
        private readonly EmailSettings _settings;

        public EmailService(IOptions<EmailSettings> settings)
        {
            _settings = settings.Value;
        }

        public void SendTransactionSuccessEmail(
            string toEmail,
            Transaction transaction,
            byte[] invoicePdf)
        {
            var message = new MimeMessage();

            message.From.Add(new MailboxAddress(
                _settings.SenderName,
                _settings.SenderEmail));

            message.To.Add(MailboxAddress.Parse(toEmail));

            message.Subject = "Bookworm - Transaction Successful";

            // Email body (TEXT)
            var body = new TextPart("plain")
            {
                Text =
                    $"Hello {transaction.User.UserName},\n\n" +
                    $"Your transaction was successful.\n\n" +
                    $"Transaction ID: {transaction.TransactionId}\n" +
                    $"Amount: ₹{transaction.TotalAmount}\n\n" +
                    $"Thank you for shopping with Bookworm!"
            };

            // PDF attachment
            var attachment = new MimePart("application", "pdf")
            {
                Content = new MimeContent(
                    new MemoryStream(invoicePdf)),
                ContentDisposition = new ContentDisposition(
                    ContentDisposition.Attachment),
                ContentTransferEncoding = ContentEncoding.Base64,
                FileName = $"invoice_{transaction.TransactionId}.pdf"
            };

            // Combine body + attachment
            var multipart = new Multipart("mixed");
            multipart.Add(body);
            multipart.Add(attachment);

            message.Body = multipart;

            using var client = new SmtpClient();
            client.Connect(
                _settings.SmtpServer,
                _settings.Port,
                SecureSocketOptions.StartTls);

            client.Authenticate(
                _settings.Username,
                _settings.Password);

            client.Send(message);
            client.Disconnect(true);
        }
    }
}

File: BOOKWORM_PRO/Bookworm/src/main/java/com/example/Services/EmailService.java
package com.example.Services;

import org.springframework.core.io.ByteArrayResource;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.mail.javamail.MimeMessageHelper;
import org.springframework.stereotype.Service;

import com.example.models.Transaction;

import jakarta.mail.internet.MimeMessage;

@Service
public class EmailService {

    private final JavaMailSender mailSender;

    public EmailService(JavaMailSender mailSender) {
        this.mailSender = mailSender;
    }

    public void sendTransactionSuccessEmail(
            String toEmail,
            Transaction transaction,
            byte[] invoicePdf) {

        try {
            MimeMessage message = mailSender.createMimeMessage();
            MimeMessageHelper helper =
                    new MimeMessageHelper(message, true);

            helper.setTo(toEmail);
            helper.setSubject("Bookworm - Transaction Successful");
            helper.setText(
                    "Hello " + transaction.getUser().getUserName() + ",\n\n" +
                    "Your transaction was successful.\n\n" +
                    "Transaction ID: " + transaction.getTransactionId() + "\n" +
                    "Amount: ₹" + transaction.getTotalAmount() + "\n\n" +
                    "Thank you for shopping with Bookworm!",
                    false
            );

            helper.addAttachment(
                    "invoice_" + transaction.getTransactionId() + ".pdf",
                    new ByteArrayResource(invoicePdf)
            );

            mailSender.send(message);

        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}